src/pages/data.test.js

```diff
@@ -0,0 +1,427 @@
+import '@testing-library/jest-dom'
+import { CustomDataProvider } from '@dhis2/app-runtime'
+import { configure, render, waitFor, within } from '@testing-library/react'
+import React from 'react'
+import { MemoryRouter, Route } from 'react-router-dom'
+import { QueryParamProvider } from 'use-query-params'
+import { ReactRouter6Adapter } from 'use-query-params/adapters/react-router-6'
+import { formatData } from '../components/view/data-workspace/requests-display/index.js'
+import { getRelativeTimeDifference } from '../components/view/data-workspace/title-bar/index.js'
+import { getReportText } from '../components/view/submit-modal/submit-modal.js'
+import { AppContext, UserContext } from '../context/index.js'
+import {
+    testDataExchange,
+    testDataExchangeSourceData,
+    testImportSummary,
+    testRequest,
+    testUserContext,
+} from '../utils/builders.js'
+import { DataPage } from './data.js'
+
+const lastAnalyticsTableSuccess = '2024-07-07T21:47:58.383'
+const serverDate = '2024-07-18T17:36:38.164'
+const contextPath = 'debug.dhis2.org/dev'
+let mockImportSummariesResponse = {}
+
+jest.mock('@dhis2/app-runtime', () => ({
+    ...jest.requireActual('@dhis2/app-runtime'),
+    useConfig: () => ({
+        baseUrl: 'https://debug.dhis2.org/dev',
+        apiVersion: '41',
+        systemInfo: {
+            lastAnalyticsTableSuccess,
+            serverDate,
+            serverTimeZoneId: 'Etc/UTC',
+            contextPath: `https://${contextPath}`,
+        },
+    }),
+    useDataEngine: jest.fn().mockReturnValue({
```

Comment:
Do you know if we can define a response in CustomDataProvider for type:CREATE to resource:aggregateDataExchanges/id/exchange and not have to mock useDataEngine? This definitely works, it's of a general question as I don't really understand what our "standard" testing procedure would be for useDataEngine

Reply:
I think i managed to get this working 👍